### DISCUSSION TITLE
--add-data folders get flattened, how to stop it from flattening it

### DISCUSSION THREAD

**Discussion by WilliamStam** [pyinstaller/pyinstaller#8175] [Help] [answered]
when using --add-data it seems to "flatten" the folders contents into that folder's root

```
PyInstaller.__main__.run(
    [
        "app/main.py",
        "--name=api",
        # "--onefile",
        "--clean",
        "--paths=app",
        "--hidden-import=app",
        "--icon=./app/static/icon.ico",
        "--add-data=./app/static/*:static",
    ]
)
```

the static folder looks like
```
static
- logos
   - logo.png
   - something.png
- elements.css
- icon.ico

```

after running the build my static folder looks like

```
static
- logo.png
- something.png
- elements.css
- icon.ico
```

i get the whole "if your paths are relative then its a bug" argument but i would think that thats for py stuff. when adding "data" im sure stuff like scripts and logos can keep their structure? (like in the css file it references a static file logos/logo.png) is there anyway for it to "keep" the structure (logos folder - this is an oversimplified example tho so "just change logos to root" isnt an answer).

**Comment by bwoodsend** [comment, marked as answer]
Just remove the wildcard. `"--add-data=./app/static:static"`

**Comment by WilliamStam** [comment]
you have GOT  to be joking... ive been struggling with that for a couple of hours. i was being a real idiot... thank you!
